src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1049,6 +1049,7 @@ protected NodeModel(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPo
         {
             inputNodes = new Dictionary<int, Tuple<int, NodeModel>>();
             outputNodes = new Dictionary<int, HashSet<Tuple<int, NodeModel>>>();
+            FindPortData(inPorts, outPorts);
```

Comment:
how was the tooltip info retrieved before this PR ?

Reply:
tooltip and name info was retrieved directly from the deserialized ports from the .dyn or .dyf - which leads to the two bugs mentioned in the description.